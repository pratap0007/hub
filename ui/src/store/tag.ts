import { types, Instance } from 'mobx-state-tree';

export const Tag = types.model('Tags', {
  id: types.number,
  name: types.identifier
});

export const TagStore = types
  .model({
    items: types.optional(types.map(Tag), {})
  })
  .actions((self) => ({
    add(item: ITag): void {
      self.items.put(item);
    }
  }))
  .views((self) => ({
    get values() {
      return Array.from(self.items.values());
    }
  }));

export type ITag = Instance<typeof Tag>;
export type ItagStore = Instance<typeof TagStore>;
