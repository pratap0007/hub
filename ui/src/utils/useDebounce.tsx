import { useEffect, useState } from 'react';
import { useMst } from '../store/root';

export const useDebounce = (value: string, delay: number) => {
  const [debouncedValue, setDebouncedValue] = useState(value);
  const { resources } = useMst();

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
      resources.setSearch(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay, resources]);
  return debouncedValue;
};
