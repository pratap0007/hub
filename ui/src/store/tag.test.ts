import { getSnapshot } from 'mobx-state-tree';
import { Tag, TagStore } from './tag';
import { assert } from './utils';

describe('Store Object', () => {
  it('can create a tag object', () => {
    const tag = Tag.create({
      id: 1,
      name: 'aws'
    });

    expect(tag.name).toBe('aws');
  });

  it('creates a tag store', (done) => {
    const store = TagStore.create({});

    const item = Tag.create({
      id: 10,
      name: 'cli'
    });

    store.add(item);

    expect(store.values[0].name).toBe('cli');

    const tags = store.items.get('cli');
    assert(tags);

    expect(tags.name).toBe('cli');
    expect(getSnapshot(store.items)).toMatchSnapshot();

    done();
  });
});
