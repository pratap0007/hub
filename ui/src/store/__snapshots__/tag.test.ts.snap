// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store Object creates a tag store 1`] = `
Object {
  "cli": Object {
    "id": 10,
    "name": "cli",
  },
}
`;
